### DISCUSSION TITLE
Extending alpinejs - access to walk() function

### DISCUSSION THREAD

**Discussion by akashchopra** [alpinejs/alpine#3176] [1. Help]
In order to learn a bit about extending alpinejs, I tried to recreate the x-if directive by adding an x-myif directive with the implementation taken from the code at https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-if.js. However, when I use it, it fails at the equivalent of https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-if.js#L28 because the walk() function is undefined.

I can't figure out how to get access to that function from a user-defined directive. Can anyone help?

**Comment by ekwoka** [comment]
You can import it from the same source. if it's not exposed in any of the context stuff (I'd check there first).

But yeah, there is a major architecture issue with Alpine where importing `alpinejs` imports a bundle of the code, instead of pointing to the actual modules themselves. So you can't always import those other internal things and have it work. I find it quite obnoxious personally. Reduces ones ability to remix.
